CRM/Upgrade/Incremental/php/FiveFiftySeven.php

```diff
@@ -29,6 +29,22 @@ class CRM_Upgrade_Incremental_php_FiveFiftySeven extends CRM_Upgrade_Incremental
    */
   public function upgrade_5_57_alpha1($rev): void {
     $this->addTask(ts('Upgrade DB to %1: SQL', [1 => $rev]), 'runSql', $rev);
+    $this->addTask('Add in Client Removed Activity Type', 'addCaseClientRemovedActivity');
+  }
+
+  public static function addCaseClientRemovedActivity() {
+    if (CRM_Core_Component::isEnabled('CiviCase')) {
```

Comment:
@seamuslee001 If I understand the previous note correctly, I think you need to do something a bit more nuanced. The above will prevent the activity type from being created if CiviCase is not enabled, which addresses the immediate concern. But if a person enables CiviCase at a later time, we need to make sure this activity is created. So perhaps we need to also add this code to the CiviCase enabling process?

Reply:
In a stock install without civicase the case-y stuff is still always inserted into the db at install time, so we should always insert this one too. How about without the `if` and set it like:

`'component_id' => CRM_Core_DAO::singleValueQuery("SELECT id FROM civicrm_component WHERE name='CiviCase'"),`